### PR TITLE
#53 refactor sync state

### PR DESCRIPTION
**Pull request by wlthomson**
Refactors sync state, including moving all non-persisted state to redux store.

### CODE REVIEW

File: src/common/services/Database/DatabaseService.ts [hunk 1]
```diff
@@ -55,24 +55,19 @@ export class DatabaseService {
...
-      host,
-      port,
-      loginPath,
-      sensorPath,
-      temperatureLogPath,
-      temperatureBreachPath,
+      loginUrl,
+      sensorUrl,
+      temperatureLogUrl,
+      temperatureBreachUrl,
```
Comment (joshxg) [inline src/common/services/Database/DatabaseService.ts:70]: maybe this should all be surrounded in a `__DEV__`?

File: src/features/Sync/SyncOutManager.ts [hunk 2]
```diff
@@ -4,147 +4,87 @@ import { DatabaseService } from '../../common/services/Database';
...
 
-    syncSensors = async (logs: SyncLog[]): Promise<void> => {
-        axios.put(this.sensorUrl, this.getSyncBody(logs), { withCredentials: true });
+    public syncSensors = async (sensorUrl: string, logs: SyncLog[]): Promise<void> => {
+        axios.put(sensorUrl, this.getSyncBody(logs), { withCredentials: true });
     }
 
-    syncTemperatureLogs = async (logs: SyncLog[]): Promise<void> => {
-        axios.put(this.temperatureLogUrl, this.getSyncBody(logs), { withCredentials: true });
+    public syncTemperatureLogs = async (temperatureLogUrl: string, logs: SyncLog[]): Promise<void> => {
```
Comment (joshxg) [inline src/features/Sync/SyncOutManager.ts:82]: I believe https://app.swaggerhub.com/apis-docs/msupply-foundation/ColdChain/1.0.1#/default/coldchain_temperatureLogV1 the endpoint is returned in the `valid` field, all the temperature logs which were saved - I think we should be returning the response and iterating over the valid logs and removing the sync out's according to those ids?

File: src/features/Sync/SyncSlice.ts [hunk 3]
```diff
@@ -1,105 +1,156 @@
...
+    payload: AuthenticateSuccessActionPayload;
 }
 
-export interface UpdatePasswordActionPayload {
-    password: string;
+export interface  SyncSensorsActionPayload {
+    sensorUrl: string;
+ } 
+
+ export interface  SyncSensorsAction {
```
Comment (joshxg) [inline src/features/Sync/SyncSlice.ts:99]: Do you think we should do a `types` file in each of the feature dirs? Or is it good in the same file?

File: src/features/Sync/SyncSlice.ts [hunk 4]
```diff
@@ -341,111 +443,135 @@ function* updatePassword({ payload: { password } }: UpdatePasswordAction): SagaI
...
-function* syncAll(): SagaIterator {
-    yield call(syncSensors);
-    yield call(syncTemperatureLogs);
-    yield call(syncTemperatureBreaches);
+function* syncAll({ payload: { loginUrl, sensorUrl, temperatureLogUrl, temperatureBreachUrl, username, password }}: SyncAllAction): SagaIterator {
+    yield put(SyncAction.authenticate(loginUrl, username, password))
+    yield delay(MILLISECONDS.TEN_SECONDS);
+    yield put(SyncAction.syncSensors(sensorUrl));
+    yield put(SyncAction.syncTemperatureLogs(temperatureLogUrl));
+    yield put(SyncAction.syncTemperatureBreaches(temperatureBreachUrl));
```
Comment (joshxg) [inline src/features/Sync/SyncSlice.ts:551]: Hm, how does this work? Is each action fully 'done' before the next?

File: src/features/Sync/SyncSlice.ts [hunk 5]
```diff
@@ -454,14 +580,28 @@ function* fetchAll(): SagaIterator {
...
+    yield takeEvery(SyncAction.authenticateFailure, authenticateFailure);
+    yield takeEvery(SyncAction.syncSensors, syncSensors);
+    yield takeEvery(SyncAction.syncSensorsSuccess, syncSensorsSuccess);
+    yield takeEvery(SyncAction.syncSensorsFailure, syncSensorsFailure);
+    yield takeEvery(SyncAction.syncTemperatureLogs, syncTemperatureLogs);
+    yield takeEvery(SyncAction.syncTemperatureLogsSuccess, syncTemperatureLogsSuccess);
+    yield takeEvery(SyncAction.syncTemperatureLogsFailure, syncTemperatureLogsFailure);
+    yield takeEvery(SyncAction.syncTemperatureBreaches, syncTemperatureBreaches);
+    yield takeEvery(SyncAction.syncTemperatureBreachesSuccess, syncTemperatureBreachesSuccess);
+    yield takeEvery(SyncAction.syncTemperatureBreachesFailure, syncTemperatureBreachesFailure);
```
Comment (joshxg) [inline src/features/Sync/SyncSlice.ts:600]: What are the plans for the failure sagas?

We might also want to `takeLatest` of a few of these

**Review by joshxg — APPROVED** [review]
Nice.

Few suggestions no show stoppers

File: src/common/services/Database/DatabaseService.ts (hunk 1 above)
Comment (wlthomson) [reply src/common/services/Database/DatabaseService.ts:70]: Resolved by #91.

File: src/features/Sync/SyncSlice.ts (hunk 3 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:99]: As discussed, agree good idea to extract into dedicated type definition files.

File: src/features/Sync/SyncSlice.ts (hunk 4 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:551]: Resolved by #57.

File: src/features/Sync/SyncSlice.ts (hunk 5 above)
Comment (wlthomson) [reply src/features/Sync/SyncSlice.ts:600]: Issued created (#92).

File: src/features/Sync/SyncOutManager.ts (hunk 2 above)
Comment (wlthomson) [reply src/features/Sync/SyncOutManager.ts:82]: Issue created (#93).

**Pull request merged**